### PR TITLE
feat(oob): receive Invitation with timeout

### PR DESCRIPTION
**Pull request by Zzocker**
Signed-off-by: Pritam Singh <pkspritam16@gmail.com>

### CODE REVIEW

File: packages/core/src/modules/oob/OutOfBandApi.ts [hunk 1]
```diff
@@ -483,7 +486,7 @@ export class OutOfBandApi {
         } else {
           // Wait until the connection is ready and then pass the messages to the agent for further processing
           this.connectionsApi
-            .returnWhenIsConnected(connectionRecord.id)
+            .returnWhenIsConnected(connectionRecord.id, config.timeoutMs ? { timeoutMs: config.timeoutMs } : undefined)
```
Comment (TimoGlastra) [inline packages/core/src/modules/oob/OutOfBandApi.ts:489]: I think we can simplify to this?

```suggestion
            .returnWhenIsConnected(connectionRecord.id, { timeoutMs: config?.timeoutMs })
```

File: packages/core/src/modules/oob/OutOfBandApi.ts [hunk 2]
```diff
@@ -75,6 +75,7 @@ export interface ReceiveOutOfBandInvitationConfig {
   autoAcceptConnection?: boolean
   reuseConnection?: boolean
   routing?: Routing
+  timeoutMs?: number
```
Comment (TimoGlastra) [inline packages/core/src/modules/oob/OutOfBandApi.ts:78]: Thinking whether we should give this another name, or whether we should set a timer for the overall action. If I set a timeoutMs I expect the whole action to have a specific timeout, but this only relates to the time we wait for the connection to be completed afterwards.

Also if autoAcceptConnection is false, this parameter won't do anything which may be a bit misleasing. 

What do you think?

Comment (Zzocker) [reply packages/core/src/modules/oob/OutOfBandApi.ts]: Yes, make sense, [FORCED PUSHED] changed the name to `acceptInvitationTimeoutMs`

File: packages/core/src/modules/oob/OutOfBandApi.ts [hunk 3]
```diff
@@ -483,7 +487,7 @@ export class OutOfBandApi {
         } else {
           // Wait until the connection is ready and then pass the messages to the agent for further processing
           this.connectionsApi
-            .returnWhenIsConnected(connectionRecord.id)
+            .returnWhenIsConnected(connectionRecord.id, { timeoutMs: timeoutMs })
```
Comment (genaris) [inline packages/core/src/modules/oob/OutOfBandApi.ts:490]:
```suggestion
            .returnWhenIsConnected(connectionRecord.id, { timeoutMs })
```

**Review by genaris — APPROVED** [review]
Just a super small suggestion but LGTM

File: packages/core/src/modules/oob/OutOfBandApi.ts (hunk 3 above)
Comment (Zzocker) [reply packages/core/src/modules/oob/OutOfBandApi.ts]: Made the suggested changes.

**Pull request merged**